### ISSUE TITLE
500 Error when invoking service request from Amazon test page

### ISSUE THREAD

**Issue by LarryEllis** [AreYouFreeBusy/AlexaSkillsKit.NET#40]
When sending an intent request from the Amazon developer test page, a 500 result occurs.

Looking at the exception, the failure is here (line 65 in SpeechletRequestEnvelope v 1.5.2):

`       return new SpeechletRequestEnvelope {
                Request = request,
                Session = Session.FromJson(json["session"].Value<JObject>()),
                Version = json["version"].Value<string>()
            };
`

The problem is the fact that the input json string does not include the _version_ field.  This, in spite of the fact that it is shown to be present in the test harness view of the Service Request.   So it seems the test harness is just not sending it.   Examination of the input request seems to bear this out.

It appears as though some parts of the library do handle a missing version value, except this portion does not.

The Amazon schema says the version field is always there, so it looks like this is a bug in Amazon's test harness.  Perhaps the library could work around the omission in a safe way.

**Comment by stefann42** [comment]
You're right, I tried to deal with this in the past, but I missed that spot. Pushed this commit https://github.com/AreYouFreeBusy/AlexaSkillsKit.NET/commit/60ef3e3d843795102112895c98d05683541c066f and will try to get a new package out soon.

Unfortunately there's no "safe" way to accommodate Amazon's test harness. They basically screwed up the Version field by not respecting their own standard and I doubt it will ever be used (although they don't seem to care much when they break 3rd party libraries so who knows).
